harvester/sources/extraction/publinova.py

```diff
@@ -154,6 +154,13 @@ def get_research_themes(cls, node):
     def get_parties(cls, node):
         return [party["name"] for party in node.get("parties", []) or []]
 
+    @classmethod
+    def get_doi(cls, node):
+        doi = node.get("doi", None)
+        if not doi:
+            return
+        return "10." + doi.split("10.", 1)[1].replace(" ", "")
```

Comment:
This will not work on a string like "crash". That string does not have any "10." in it so you'll get an IndexError when trying to access element 1. You could try to catch the IndexError or switch to Python's [string.index](https://www.programiz.com/python-programming/methods/string/index) and catch its ValueError. Another route would be to use regex and to return None when the match is None.

It safer to use trim function instead of replace(" ", ""). Trim will remove all white space (including tabs and newlines) and only at the beginning or end of the string. Encoding spaces to "+" is more inline with HTTP than stripping them.